Fix: get Mudlet starting after installation on Windows

SlySven opened the pull request:
This includes additional diagnostic code that will put a `[ WARN ] -` message up in the main console of the first profile opened after Mudlet is started with a command line argument that is not recognised. **It no longer aborts the run in that case.**

This should close #6500 as it appears that that is being caused by a command line option from the installer and since #6470 Mudlet would abort and show the `-h`/`--help` output should it get a command line argument that it doesn't recognise.

This is an alternative to #6582 - that removes the non-zero exit code and reports the parsing error message to the OS command line (as before) rather than to the game (which this PR does) which means it may not be visible to the end user.

SlySven committed:
Fix: get Mudlet starting after installation on Windows

This includes additional diagnostic code that will put a `[ WARN ] -`
message up in the main console of the first profile opened after Mudlet
is started with a command line argument that is not recognised.

This should close #6500 as it appears that that is being caused by a
command line option from the installer and since #6470 Mudlet would abort
and show the `-h`/`--help` output should it get a command line argument
that it doesn't recognise.

This is an alternative to #6582 which instead removes the non-zero exit
code and reports the parsing error message to the OS command line rather
than to the game which means it may not be visible to the user.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: typo that I didn't spot

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
In use - Mudlet was invoked (on GNU/Linux) with **`mudlet --badOption --anotherThing rubbish`**:

![image](https://user-images.githubusercontent.com/6163092/221425724-746d0faf-fd90-4191-aaf6-2beadcf46921.png)

SlySven commented:
Assigning @demonnic as I am not sure Vadi is around and he covers both groups needed for approval.

vadi2 commented:
Thank you for working on the option, but I prefer https://github.com/Mudlet/Mudlet/pull/6582 because it is far more consise

SlySven committed:
Fix: remember an application exits with a zero exit code on success!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: reduce the size of the changes

Peer review thought this was less concise than the alternative so I have
tried to reduce the number of changes.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: polish some strings mentioning "version"

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
So do we want to go just with @atari2600tim 's PR or will this one (which I have shaved down a bit but it still does a bit more) get a :heavy_check_mark: ?

vadi2 commented:
Out of the two I prefer the https://github.com/Mudlet/Mudlet/pull/6582 solution because in this case a really simple one will do the job for a simple problem. But, this work was not wasted because it gave us some implementation options to compare :)

SlySven committed:
Merge branch 'development' into Fix_getMudletStartingAfterInstallationOnWindows

SlySven commented:
Okay, closing it...
